src/storage/data_table.cpp

```diff
@@ -246,6 +274,189 @@ bool DataTable::Delete(transaction::TransactionContext *const txn, const TupleSl
   return true;
 }
 
+void DataTable::WriteSchemaMessage(std::ofstream &outfile, std::unordered_map<col_id_t, int64_t> *dictionary_ids,
+                                   flatbuffers::FlatBufferBuilder *flatbuf_builder) const {
+  RawBlock *block = blocks_.front();
+  const BlockLayout &layout = accessor_.GetBlockLayout();
+  ArrowBlockMetadata &metadata = accessor_.GetArrowBlockMetadata(block);
+  std::vector<flatbuffers::Offset<flatbuf::Field>> fields;
+  int64_t dictionary_id = 0;
+  for (col_id_t col_id : layout.AllColumns()) {
+    ArrowColumnInfo &col_info = metadata.GetColumnInfo(layout, col_id);
+    // Current column name is meaningless
+    auto name = flatbuf_builder->CreateString("TestCol" + std::to_string(!col_id));
+    flatbuf::Type type;
+    flatbuffers::Offset<void> type_offset;
+    flatbuffers::Offset<flatbuf::DictionaryEncoding> dictionary = 0;
+    if (!layout.IsVarlen(col_id) || col_info.Type() == ArrowColumnType::FIXED_LENGTH) {
+      uint8_t byte_width = accessor_.GetBlockLayout().AttrSize(col_id);
+      type = flatbuf::Type_FixedSizeBinary;
+      type_offset = flatbuf::CreateFixedSizeBinary(*flatbuf_builder, byte_width).Union();
+    } else {
+      switch (col_info.Type()) {
+        case ArrowColumnType::DICTIONARY_COMPRESSED:
+          dictionary = flatbuf::CreateDictionaryEncoding(
+              *flatbuf_builder, dictionary_id, flatbuf::CreateInt(*flatbuf_builder, 8 * sizeof(uint64_t), true), false);
+          (*dictionary_ids)[col_id] = dictionary_id++;
+        case ArrowColumnType::GATHERED_VARLEN:  // fall through
+          type = flatbuf::Type_LargeBinary;
+          type_offset = flatbuf::CreateLargeBinary(*flatbuf_builder).Union();
+          break;
+        default:
+          throw std::runtime_error("unexpected control flow");
+      }
+    }
+    std::vector<flatbuffers::Offset<org::apache::arrow::flatbuf::Field>> fake_children;
```

Comment:
nit: Can you add a comment explaining why this is called `fake_children`. Something like "We don't care about the children because...."